lib/httpMock.js

```diff
@@ -15,15 +15,13 @@ function mockTemplate() {
 
         $provide.decorator('$http', ['$delegate', '$q', '$injector', function($http, $q, $injector) {
 
-        var interceptors = $httpProvider.interceptors;
+        var interceptors = [];
 
-        function getInterceptor(interceptorExpression) {
-            if (angular.isString(interceptorExpression)) {
-                return $injector.get(interceptorExpression);
-            } else {
-                return $injector.invoke(interceptorExpression);
-            }
-        }
+        angular.forEach($httpProvider.interceptors, function(interceptorExpression) {
+            var interceptor = angular.isString(interceptorExpression) ?
+                $injector.get(interceptorExpression) : $injector.invoke(interceptorExpression);
+            interceptors.push(interceptor);
+        });
```

Comment:
I think it'd be cleaner to keep the if/else block rather than use a ternary operator.